src/golang/cmd/executor/executor/dynamic_teardown.go

```diff
@@ -41,6 +41,11 @@ func (ex *DynamicTeardownExecutor) Run(ctx context.Context) error {
 	}
 
 	dynamicIntegration := dynamicIntegrations[0]
+
+	if err := dynamic.ResyncClusterState(ctx, &dynamicIntegration, ex.IntegrationRepo, ex.Database); err != nil {
```

Comment:
Should all DB access be in a txn to avoid concurrency issue, if this method can be called concurrently?